ci/tests/arakoon/extend_promote_demote_test.py

```diff
@@ -447,3 +449,48 @@ def ovs_3671_validate_archiving_of_existing_arakoon_data_on_create_and_extend_te
             for filename in files_to_create:
                 assert client.file_exists(filename) is False, 'File {0} is missing'.format(filename)
             TestArakoon.verify_arakoon_structure(client, cluster_name, False, False)
+
+    @staticmethod
+    def ovs_4509_validate_arakoon_collapse_test():
+        """
+        Validate arakoon collapse
+        """
+        node_ips = [sr.ip for sr in GeneralStorageRouter.get_storage_routers()]
+        node_ips.sort()
+        arakoon_conf_file = '/etc/init/ovs-arakoon-ovsdb.conf'
+        etcd_config = 'etcd://127.0.0.1:2379/ovs/arakoon/ovsdb/config'
+        tlog_location = '/opt/OpenvStorage/db/arakoon/ovsdb/tlogs'
+        first_ip = node_ips[0]
+        root_client = SSHClient(first_ip, username='root')
+        # read_conf_settings
+        conf_contents = root_client.file_read(arakoon_conf_file)
+        for split_item in conf_contents.splitlines()[-1].split():
+            if 'etcd' in split_item:
+                etcd_config = split_item
+        # read_tlog_dir
+        etcd_arakoon_conf = re.findall('\d+|\D+', etcd_config)[-1]
+        etcd_contents = root_client.run('etcdctl get {0}'.format(etcd_arakoon_conf))
+        for line in etcd_contents.splitlines():
+            if 'tlog_dir' in line:
+                tlog_location = line.split()[-1]
+
+        no_of_tlogs = 0
+        for file_name in root_client.file_list(tlog_location):
+            if file_name.endswith('tlx') or file_name.endswith('tlog'):
+                no_of_tlogs += 1
+        old_headdb_timestamp = root_client.run('stat --format=%Y {0}/{1}'.format(tlog_location, 'head.db'))
+        if no_of_tlogs <= 2:
+            # run_arakoon_benchmark
+            benchmark_command = 'arakoon --benchmark -n_clients 1 -max_n 10_000 -config {0}'.format(etcd_config)
+            root_client.run(benchmark_command)
+        # run_collapse
```

Comment:
It would be nice it we would verify if the collapse works for all arakoons we use on opnvestorage ... you can find some inspiration in the collapse_arakoon method of the ScheduledTaskController ...
This will allow you to make this code more robust (i.e. less file retrieval and parsing)